zerver/tests/test_thumbnail.py

```diff
@@ -0,0 +1,207 @@
+        fp.name = "zulip.jpeg"
+
+        result = self.client_post("/json/user_uploads", {'file': fp})
+        self.assert_json_success(result)
+        json = ujson.loads(result.content)
+        self.assertIn("uri", json)
+        uri = json["uri"]
+        base = '/user_uploads/'
+        self.assertEqual(base, uri[:len(base)])
+
+        # Without specifying size we should get the original image
+        # (Known as 0x0 configuration in thumbor)
+        result = self.client_get("/thumbnail" + uri)
+        self.assertEqual(result.status_code, 302, result)
+        base = '/thumbor/unsafe/0x0/smart/files/'
+        self.assertEqual(base, result.url[:len(base)])
+
+        # Test with size supplied as a query parameter.
+        result = self.client_get("/thumbnail%s?size=480x360" % (uri))
+        self.assertEqual(result.status_code, 302, result)
+        base = '/thumbor/unsafe/480x360/smart/files/'
+        self.assertEqual(base, result.url[:len(base)])
+
+        # Test with a unicode filename.
+        fp = StringIO("zulip!")
+        fp.name = "μένει.jpg"
+
+        result = self.client_post("/json/user_uploads", {'file': fp})
+        self.assert_json_success(result)
+        json = ujson.loads(result.content)
+        self.assertIn("uri", json)
+        uri = json["uri"]
+        result = self.client_get("/thumbnail" + uri)
+        self.assertEqual(result.status_code, 302, result)
+        base = '/thumbor/unsafe/0x0/smart/files/'
+        self.assertEqual(base, result.url[:len(base)])
+
+        # Test with another user trying to access image using thumbor.
+        self.logout()
+        self.login(self.example_email("iago"))
+        result = self.client_get("/thumbnail" + uri)
+        self.assertEqual(result.status_code, 403, result)
+        self.assert_in_response("You are not authorized to view this file.", result)
+
+    @override_settings(THUMBOR_HOST='127.0.0.1:9995')
+    def test_with_static_files(self):
+        # type: () -> None
+        self.login(self.example_email("hamlet"))
+        uri = '/static/images/cute/turtle.png'
+        result = self.client_get("/thumbnail" + uri)
+        self.assertEqual(result.status_code, 302, result)
+        self.assertEqual(uri, result.url)
+
+    def test_with_thumbor_disabled(self):
+        # type: () -> None
+        self.login(self.example_email("hamlet"))
+        fp = StringIO("zulip!")
+        fp.name = "zulip.jpeg"
+
+        result = self.client_post("/json/user_uploads", {'file': fp})
+        self.assert_json_success(result)
+        json = ujson.loads(result.content)
+        self.assertIn("uri", json)
+        uri = json["uri"]
+        base = '/user_uploads/'
+        self.assertEqual(base, uri[:len(base)])
+
+        with self.settings(THUMBOR_HOST=''):
+            result = self.client_get("/thumbnail" + uri)
+        self.assertEqual(result.status_code, 302, result)
+        base = '/user_uploads/'
+        self.assertEqual(base, result.url[:len(base)])
```

Comment:
let's verify the entire URL in these tests; I think that'd be more robust to potential future bugs.